fastsync/event: emit fastsync status event when switching consensus/fastsync (#6619)
tendermint/tendermint: docs/tendermint-core/fast-sync.md

If we're lagging sufficiently, we should go back to fast syncing, but
this is an [open issue](https://github.com/tendermint/tendermint/issues/129).

> ## The Fast Sync event

Reviewer: ```suggestion
## The Fast Sync event

```

> When the tendermint blockchain core launches, it might switch to the `fast-sync`

Reviewer: ```suggestion
When the Tendermint blockchain launches, it might switch to the `fast-sync`
```

> mode to catch up the states to the current network best height. the core will emits

Reviewer: ```suggestion
mode to catch up the states to the current network best height. The blochchain reactor will emit
```

a fast-sync event to expose the current status and the sync height. Once it catched

> the network best height, it will switches to the state sync mechanism and then emit

Reviewer: ```suggestion
the network best height, it will switches to the state sync mechanism and then emits
```
